src/jorphan/src/main/java/org/apache/jorphan/math/StatCalculator.java

```diff
@@ -38,9 +40,9 @@ public abstract class StatCalculator<T extends Number & Comparable<? super T>> {
     // We use a TreeMap because we need the entries to be sorted
 
     // Running values, updated for each sample
-    private double sum = 0;
+    private BigDecimal sum = BigDecimal.ZERO;
 
-    private double sumOfSquares = 0;
+    private BigDecimal sumOfSquares = BigDecimal.ZERO;
 
```

Comment:
Does this change belong to "support nanoseconds" PR?
If you want to improve precision of `StatCalculator`, I would suggest creating a separate PR and discuss it there.

Have you checked if the change from `double` to `BigDecimal` produces no significant performance regressions?